{{cookiecutter.profile_name}}/lsf_submit.py

```diff
@@ -0,0 +1,187 @@
+                                    "{rule_name}.{wildcards_str}".format(rule_name=self.rule_name,
+                                                                         wildcards_str=self.wildcards_str))
+
+    @property
+    def jobid(self):
+        if self.is_group_jobtype:
+            return self.job_properties.get("jobid", "").split("-")[0]
+        else:
+            return self.job_properties.get("jobid")
+
+    @property
+    def logdir(self):
+        return Path(self.cluster.get("logdir", CookieCutter.get_log_dir()))
+
+    @property
+    def outlog(self):
+        return self.logdir / "{jobid}_{random_string}.out".format(jobid=self.jobid, random_string=self.random_string)
+
+    @property
+    def errlog(self):
+        return self.logdir / "{jobid}_{random_string}.err".format(jobid=self.jobid, random_string=self.random_string)
+
+    @property
+    def jobinfo_cmd(self):
+        return '-o "{out_log}" -e "{err_log}" -J "{jobname}"'.format(out_log=self.outlog, err_log=self.errlog, jobname=self.jobname)
+
+    @property
+    def queue(self):
+        return self.cluster.get("queue", "")
+
+    @property
+    def queue_cmd(self):
+        return "-q {}".format(self.queue) if self.queue else ""
+
+    @property
+    def cluster_cmd(self):
+        return self._cluster_cmd
+
+    @property
+    def submit_cmd(self):
+        return "bsub {resources} {job_info} {queue} {cluster} {jobscript}".format(
+                    resources=self.resources_cmd,
+                    job_info=self.jobinfo_cmd,
+                    queue=self.queue_cmd,
+                    cluster=self.cluster_cmd,
+                    jobscript=self.jobscript,
+        )
+
+    def _create_logdir(self):
+        OSLayer.mkdir(self.logdir)
+
+    def _remove_previous_logs(self):
+        OSLayer.remove_file(self.outlog)
+        OSLayer.remove_file(self.errlog)
+
+    def _submit_cmd_and_get_external_job_id(self):
+        output_stream, error_stream = OSLayer.run_process_and_get_output_and_error_stream(self.submit_cmd)
+        match = re.search(r"Job <(\d+)> is submitted", output_stream)
+        jobid = match.group(1)
+        return int(jobid)
+
+    def _get_information_to_status_script(self, external_job_id: int):
+        return "{external_job_id} {outlog}".format(
+            external_job_id=external_job_id,
+            outlog=self.outlog
+        )
+
+    def submit(self):
+        self._create_logdir()
+        self._remove_previous_logs() # we could be very unlucky of having the same 64-length random string with the same jobid
+        try:
+            external_job_id = self._submit_cmd_and_get_external_job_id()
+            information_to_status_script = self._get_information_to_status_script(external_job_id)
+            OSLayer.print(information_to_status_script)
+        except subprocess.CalledProcessError as error:
+            raise error
+        except AttributeError as error:
+            raise error
```

Comment:
If you are `except`ing these errors, only to `raise` them then  you dont need to `except` them do you?

Reply:
True, I added some more verbosity to errors when we catch them at https://github.com/leoisl/snakemake-lsf/commit/cb1bce24ef9ea0ea2e0126965382953fc9dc4aaa - it could be interesting to have as much verbosity as possible to understand issues

Reply:
You could catch it and rather than `print` you can `raise` a custom error called `BsubInvocationError`?

Reply:
You are right. Done in https://github.com/leoisl/snakemake-lsf/commit/44b0ab5bffc3561c211571ea6a9511e798564161